src/Middlewares/OptimizeImages.php

```diff
@@ -15,7 +15,9 @@ public function handle($request, Closure $next)
         collect($request->allFiles())
             ->flatten()
             ->each(function (UploadedFile $file) use ($optimizerChain) {
-                $optimizerChain->optimize($file->getPathname());
+                if ($file->isValid()) {
```

Comment:
Move this check to a `filter` operation right before the `each`

Reply:
Yeah, you're right we should filter first rather than checking every file. I'll do it!